src/workerd/server/workerd.capnp

```diff
@@ -330,6 +330,8 @@ struct Worker {
       # TODO(someday): dispatch, analyticsEngine, other new features
     }
 
+    wrapWith @14 :Text;
```

Comment:
I think that a single custom binding might need to wrap multiple inner bindings sometimes. For example, you may need a service binding that the event is sent to, plus a JSON binding containing configuration for the wrapper.

So instead of a single `wrapWith` field I would define a new binding type in the above union:

```
custom @14 :CustomBinding;
```

```
struct CustomBinding {
  innerBindings @0 :List(Binding);
  wrapWith @1 :Text;
}
```

Reply:
implemented with different name: `wrapped @14 :WrappedBinding;` and updated the sample.

Then the config looks like:

```
  bindings = [
    ( 
      name = "server",
      wrapped = (
        innerBindings = [ (name = "target", service = "server") ],
        wrapWith = "jsonRpc",
      )
    )
  ],
```

and `wrap` function has access to more than one:

```
export function wrap(env) {
  return async (json) => {
    const request = new Request("http://fake-host", { method: "POST", body: JSON.stringify(json) } );
    return await (await env.target.fetch(request)).json();
  }
}
```

Note that inner bindings won't be accessible outside.